src/BenchmarksApps/TodosApi/TodoApi.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Npgsql;
+using TodosApi;
+
+namespace Microsoft.AspNetCore.Routing;
+
+internal static class TodoApi
+{
+    public static RouteGroupBuilder MapTodoApi(this IEndpointRouteBuilder routes)
+    {
+        var group = routes.MapGroup("/api/todos");
+
+        group.MapGet("/", (NpgsqlDataSource db, CancellationToken ct) => db.QueryAsync<Todo>("SELECT * FROM Todos", ct))
+            .WithName("GetAllTodos");
```

Comment:
What does the `WithName` do? Do we need it?

Reply:
It sets the route name and endpoint name metadata which can be used for route resolution, URL generation, Open API metadata, etc. Not strictly required in this app at the moment as we aren't doing any of those things.